package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { FlatList, TouchableOpacity, TouchableOpacityProps } from 'react-native';
+import {
+  isSuggestionCommand,
+  isSuggestionEmoji,
+  isSuggestionUser,
+  Suggestion,
+  SuggestionComponentType,
+  Suggestions,
+  useSuggestionsContext,
+} from '../../contexts/suggestionsContext/SuggestionsContext';
+import type { DefaultUserType } from '../../types/types';
+
+import type { AutoCompleteSuggestionHeaderProps } from './AutoCompleteSuggestionHeader';
+import type { AutoCompleteSuggestionItemProps } from './AutoCompleteSuggestionItem';
+import type { SuggestionsContextValue } from '../../contexts/suggestionsContext/SuggestionsContext';
+import { useTheme } from '../../contexts/themeContext/ThemeContext';
+
+type AutoCompleteSuggestionListComponentProps<Us extends DefaultUserType = DefaultUserType> = {
+  active: boolean;
+  headerProps: AutoCompleteSuggestionHeaderProps;
+  suggestions: Suggestions<Us>;
+  type: SuggestionComponentType;
+};
+
+export type AutoCompleteSuggestionListPropsWithContext<
+  Us extends DefaultUserType = DefaultUserType,
+> = AutoCompleteSuggestionListComponentProps<Us> &
+  Pick<SuggestionsContextValue, 'AutoCompleteSuggestionHeader' | 'AutoCompleteSuggestionItem'>;
+
+const SuggestionsItem: React.FC<TouchableOpacityProps> = (props) => {
+  const { children, ...touchableOpacityProps } = props;
+  return <TouchableOpacity {...touchableOpacityProps}>{children}</TouchableOpacity>;
+};
+
+SuggestionsItem.displayName = 'SuggestionsHeader{messageInput{suggestions}}';
+
+export const AutoCompleteSuggestionListWithContext = <Us extends DefaultUserType = DefaultUserType>(
+  props: AutoCompleteSuggestionListPropsWithContext<Us>,
+) => {
+  const {
+    AutoCompleteSuggestionHeader,
+    AutoCompleteSuggestionItem,
+    headerProps: { type, value },
```

Comment:
Can we drop `headerProps` and instead simply pass value as a prop here, since we already have `type` being passed explicitly? Also `value` may not be an appropriate name ... maybe `queryText`?

Reply:
@khushal87 How was this resolved since you've marked it as such? Did you make any decision about this?

Reply:
Yes we discussed this in morning during some debugging.

Reply:
Its coming in next commit as mentioned by Khushal :)

Reply:
That's a bit confusing, could we maybe agree to only resolve conversations after they have been resolved rather than doing it upfront? Things get "out of sight out of mind", and confusing to anybody who hasn't been explicitly given that information/context.